requirements/prod.txt

```diff
@@ -19,7 +19,7 @@ packaging
 pycryptodome>=3.18.0,<4.0.0
 pyyaml
 password_strength
-pydantic!=1.8,!=1.8.1,<2.0.0,>=1.7.4
+pydantic==2.7.1
```

Comment:
```suggestion
pydantic>=2.7.1,<3
```
in libraries, such like `bittensor` package, it is better not to be very strict with dependency versions when as it very easy to get conflicts in the application. pydantic is using semver, restricting just major version will be fine.

This is extra problematic if you get a security update in one of dependencies, but some library opted to do a strict `==` and you have to do some hacks just to update minor/bugfix version

Reply:
Agreed. I will increase the scope for the versioning. 

@thewhaleking, I will look into the changes between 2.0 and 2.7, and if everything is kosher I will increase scope to start with 2.0.

Reply:
@thewhaleking I believe you are right, 

https://docs.pydantic.dev/latest/version-policy/

I will do the update.